course/src/isometry.cc

```diff
@@ -0,0 +1,131 @@
+#include <cmath>
+#include <cstdint>
+#include "isometry.h"
+
+namespace ekumen {
+namespace math {
+
+double& Vector3::operator [] (const int index) const {
+    if(index > Vector3::MAX_INDEX) {
+        //throw error
+    }
+    return *(data_[index]);
+}
+
+double& Vector3::operator [] (const int index) {
+    if(index > Vector3::MAX_INDEX) {
+        //throw error
+    }
+    return *(data_[index]);
```

Comment:
Avoid going through pointers. As my previous comment suggests, when you do this sometimes the compiler cannot detect when your're doing something wrong.

I've been hit by this kind of error. More than once. Do better than me.